actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -2033,7 +2033,7 @@ def add_route(action, controller, options, _path, to, via, formatted, anchor, op
               name_for_action(options.delete(:as), action)
             end
 
-            path = Mapping.normalize_path URI::RFC2396_PARSER.escape(path), formatted
+            path = Mapping.normalize_path ::URI::DEFAULT_PARSER.escape(path), formatted
```

Comment:
Was this changed intended?

Reply:
It needs to adopt `::URI` (because it wants to avoid our here-introduced local `URI`), but the switch to `RFC2396_PARSER` is a different recently-landed change.

```suggestion
            path = Mapping.normalize_path ::URI::RFC2396_PARSER.escape(path), formatted
```